### ISSUE TITLE
Bulk-Crafting should behave differently with Extensions

### ISSUE THREAD

**Issue by LcorpOfficial** [WolfyScript/CustomCrafting#35]
### The Problem

When using extensions to recipes, I have a command that runs after crafting an item. When shift-clicking, it still only runs once upon the click, instead of how many times there are crafted items. This is viewed as a bug in my eyes as i have it set to give alchemy xp in mcmmo via command when crafting wands, which doesn’t work right when crafting many vs one

### Solution

Either have an option for bulk-crafting extensions vs. single in the config, or integrate MCMMO xp as a reward extension natively via API and have all results considered when providing extended results (receive the xp 64 times for a 64x bulk craft, etc)

### Alternatives

Unsure if there is a way to prevent bulk crafting for a specific recipe as well, but have considered that as an alternative.

### Original support message: 
https://discord.com/channels/477026331096514571/498358213842960394/921737778163753031

### Basic Wand Recipe
https://pastebin.com/H8rAX42v

### Submit issue

- [X] searched for and ensured there isn't already an open issue regarding this,
- [X] are running the latest version of your server software (Spigot, Paper, etc.), and plugins (WolfyUtilities, CustomCrafting).

**Comment by WolfyScript** [comment]
I think an option would be best. The default would be to just run the extension once.
Now I need to think about a way to implement that feature. It should probably be part of the crafting recipe config.

**Comment by LcorpOfficial** [comment]
I mean you catch the method for crafting already, correct? (noting previous conversations where we discussed the Itemstack variances from methods) 

Couldn't it be as simple as having a nested attribute for single crafting only being a default with the current format (to not break current setups and would function as it currently does) and add one for bulk-crafted?


### example of single crafting RE(current): 

```
      "tags" : [ ],
    "extensions" : [ {
      "key" : "customcrafting:command",
      "outerRadius" : [ 0.0, 0.0, 0.0 ],
      "innerRadius" : [ 0.0, 0.0, 0.0 ],
      "consoleCommands" : [ "addxp %player_name% Alchemy 150" ],
      "playerCommands" : [ ],
      "nearPlayer" : false,
      "nearWorkstation" : false

```
###       example of RE (bulk crafting included):
Bulk methods would work (at least with commands) with a config option added called `"bulkMethod"`.

options would be:
`each` - Result would be awarded for each item coming from the bulk craft
`once` - Result would run only once; current behavior when bulk-crafting


tldr:
if crafted in bulk, check  `"bulkMethod"` to see what do with this RE

Do command REs only do this or do all?
      
```
      "tags" : [ ],
    "extensions" : [ {
      "key" : "customcrafting:command",
      "outerRadius" : [ 0.0, 0.0, 0.0 ],
      "innerRadius" : [ 0.0, 0.0, 0.0 ],
      "consoleCommands" : [ "addxp %player_name% Alchemy 150" ],
      "playerCommands" : [ ],
      "bulkMethod":["each"]
      "nearPlayer" : false,
      "nearWorkstation" : false
```

**Comment by WolfyScript** [comment]
Ah, that is also an idea, to have the setting inside the extension itself.  
I will see how it can be implemented.

**Comment by LcorpOfficial** [comment]
seems the least destructive way imo but I have faith you'd find the best way :D
